### PR TITLE
Code optimization for rpc-enhancement module

### PR DESCRIPTION
**Pull request by DerekYRC**
## PR Type

Refactoring

## Describe what this PR does for and how you did.

Code optimization for rpc-enhancement module

## Checklist

- [x] Add copyright holder at the beginning of .class file if it is new.
- [x] Add information of this PR to CHANGELOG.md in root of project.
- [x] All junit tests passing.
- [x] Coverage from `Codecov Report` should not decrease.

## Checklist (Optional)

- [x] Will Pull Request to branch of 2020.0 and 2021.0.
- [x] Add documentation in javadoc in code or comment below the PR if necessary.
- [x] Add your name as @author to the beginning of .class file.

### CODE REVIEW

File: spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/DefaultEnhancedFeignPluginRunner.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.tencent.cloud.rpc.enhancement.feign;
```
Comment (SkyeBeFreeman) [inline spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/DefaultEnhancedFeignPluginRunner.java:1]: Add LICENCE.

File: spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/EnhancedFeignPluginRunner.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.tencent.cloud.rpc.enhancement.feign;
```
Comment (SkyeBeFreeman) [inline spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/EnhancedFeignPluginRunner.java:1]: Add LICENCE.

File: spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/DefaultEnhancedFeignPluginRunner.java (hunk 1 above)
Comment (DerekYRC) [reply spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/DefaultEnhancedFeignPluginRunner.java:19]: done

File: spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/EnhancedFeignPluginRunner.java (hunk 2 above)
Comment (DerekYRC) [reply spring-cloud-tencent-rpc-enhancement/src/main/java/com/tencent/cloud/rpc/enhancement/feign/EnhancedFeignPluginRunner.java:19]: done

**Review by lepdou — APPROVED** [review]

**Comment by SkyeBeFreeman** [comment]
@DerekYRC 这里验证了一下resttemplate调用，发现无法进行服务发现。通过debug发现是这里在已经有`RestTemplateCustomizer`类型的bean就不会创建了，因此无法使用`RestTemplateCustomizer`的方式向resttemplate添加`EnhancedRestTemplateReporter`

image

**Comment by DerekYRC** [comment]
@SkyeBeFreeman  疏忽了，今晚修复下。只是不明白为什么restTemplateCustomizer方法上使用了ConditionalOnMissingBean注解，却在loadBalancedRestTemplateInitializerDeprecated方法中注入RestTemplateCustomizer列表。
image

**Comment by SkyeBeFreeman** [comment]
> @SkyeBeFreeman 疏忽了，今晚修复下。只是不明白为什么restTemplateCustomizer方法上使用了ConditionalOnMissingBean注解，却在loadBalancedRestTemplateInitializerDeprecated方法中注入RestTemplateCustomizer列表。 image

我也感觉很奇怪，感觉这么做，customizer就不能做其他使用了=。=

**Pull request merged**